src/main/drivers/accgyro/accgyro_spi_lsm6dsv16x.c

```diff
@@ -0,0 +1,267 @@
+
+        int16_t *accData = (int16_t *)acc->gyro->dev.rxBuf;
+
+        acc->ADCRaw[X] = accData[1];
+        acc->ADCRaw[Y] = accData[2];
+        acc->ADCRaw[Z] = accData[3];
+        break;
+    }
+
+    case GYRO_EXTI_INT_DMA:
+    {
+        // If read was triggered in interrupt don't bother waiting. The worst that could happen is that we pick
+        // up an old value.
+
+        // This data was read from the gyro, which is the same SPI device as the acc
+        int16_t *accData = (int16_t *)acc->gyro->dev.rxBuf;
+
+        acc->ADCRaw[X] = accData[4];
+        acc->ADCRaw[Y] = accData[5];
+        acc->ADCRaw[Z] = accData[6];
+        break;
+    }
+
+    case GYRO_EXTI_INIT:
+    default:
+        break;
+    }
+
+    return true;
+}
+
+bool lsm6dsv16xSpiAccDetect(accDev_t *acc)
+{
+    if (acc->mpuDetectionResult.sensor != LSM6DSV16X_SPI) {
+        return false;
+    }
+
+    acc->initFn = lsm6dsv16xAccInit;
+    acc->readFn = lsm6dsv16xAccReadSPI;
+
+    return true;
+}
+
+void lsm6dsv16xGyroInit(gyroDev_t *gyro)
+{
+    const extDevice_t *dev = &gyro->dev;
+
+    spiSetClkDivisor(dev, spiCalculateDivider(LSM6DSV16X_MAX_SPI_CLK_HZ));
+
+    // Perform a software reset
+    spiWriteReg(dev, LSM6DSV_CTRL3, LSM6DSV_CTRL3_SW_RESET);
+
+    // Select high-accuracy ODR mode 1 before leaving power-off mode
+    spiWriteReg(dev, LSM6DSV_HAODR_CFG,
+                LSM6DSV_ENCODE_BITS(LSM6DSV_HAODR_MODE1,
+                                    LSM6DSV_HAODR_CFG_HAODR_SEL_MASK,
+                                    LSM6DSV_HAODR_CFG_HAODR_SEL_SHIFT));
+
+    // Enable the gyro in high accuracy mode at 8kHz
+    spiWriteReg(dev, LSM6DSV_CTRL2,
+                LSM6DSV_ENCODE_BITS(LSM6DSV_CTRL2_OP_MODE_G_HIGH_ACCURACY,
+                                    LSM6DSV_CTRL2_OP_MODE_G_MASK,
+                                    LSM6DSV_CTRL2_OP_MODE_G_SHIFT) |
+                LSM6DSV_ENCODE_BITS(LSM6DSV_CTRL2_ODR_G_8000HZ,
+                                    LSM6DSV_CTRL2_ODR_G_MASK,
+                                    LSM6DSV_CTRL2_ODR_G_SHIFT));
+
+    // Enable 2000 deg/s sensitivity
+    spiWriteReg(dev, LSM6DSV_CTRL6,
+                LSM6DSV_ENCODE_BITS(LSM6DSV_CTRL6_FS_G_BW_407HZ,
+                                    LSM6DSV_CTRL6_LPF1_G_BW_MASK,
+                                    LSM6DSV_CTRL6_LPF1_G_BW_SHIFT) |
+                LSM6DSV_ENCODE_BITS(LSM6DSV_CTRL6_FS_G_2000DPS,
+                                    LSM6DSV_CTRL6_FS_G_MASK,
+                                    LSM6DSV_CTRL6_FS_G_SHIFT));
+
+    // Autoincrement register address when doing block SPI reads and update continuously
+    spiWriteReg(dev, LSM6DSV_CTRL3, LSM6DSV_CTRL3_IF_INC);
+
+    // Generate pulse on interrupt line, not requiring a read to clear
+    // TODO this pulse lasts 66us followed by a low of 66us, so we get 132us cycle time, not 125us
+    spiWriteReg(dev, LSM6DSV_CTRL4, LSM6DSV_CTRL4_DRDY_PULSED);
+
+    gyro->scale = GYRO_SCALE_2000DPS;
```

Comment:
Interesting observation. Thank you.

2000 / 0x8000 = 0.061 deg/LSB. So that would be the ideal.

From https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Datasheet1.pdf I see that the MPU6000 has a figure of 0.066, for example, but no other driver set this figure according to their datasheet.

`#define GYRO_SCALE_2000DPS (2000.0f / (1 << 15))`

That said I've set the scale for this device according to the datasheet and will raise a PR to improve calibration for all others.

On a similar note the acc has a sensitivity of 0.488 mg/LSB with FS = ±16 g. 0x8000 * 0.488/1000 = 15.99, so that is not in need of special consideration.